homeassistant/components/vacuum/roomba.py

```diff
@@ -0,0 +1,269 @@
+ATTR_BIN_PRESENT = 'bin_present'
+ATTR_CLEANING_TIME = 'cleaning_time'
+ATTR_CLEANED_AREA = 'cleaned_area'
+ATTR_ERROR = 'error'
+ATTR_POSITION = 'position'
+ATTR_SOFTWARE_VERSION = 'software_version'
+
+CONF_CERT = 'certificate'
+CONF_CONTINUOUS = 'continuous'
+
+DEFAULT_CERT = '/etc/ssl/certs/ca-certificates.crt'
+DEFAULT_CONTINUOUS = True
+DEFAULT_NAME = 'Roomba'
+
+ICON = 'mdi:roomba'
+PLATFORM = 'roomba'
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_CERT, default=DEFAULT_CERT): cv.string,
+    vol.Optional(CONF_CONTINUOUS, default=DEFAULT_CONTINUOUS): cv.boolean,
+}, extra=vol.ALLOW_EXTRA)
+
+SUPPORT_ROOMBA = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_PAUSE | \
+                 SUPPORT_STOP | SUPPORT_RETURN_HOME | SUPPORT_BATTERY | \
+                 SUPPORT_STATUS | SUPPORT_SEND_COMMAND
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the iRobot Roomba vacuum cleaner platform."""
+    from roomba import Roomba
+    if PLATFORM not in hass.data:
+        hass.data[PLATFORM] = {}
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    certificate = config.get(CONF_CERT)
+    continuous = config.get(CONF_CONTINUOUS)
+
+    # Create handler
+    roomba = Roomba(
+        address=host,
+        blid=username,
+        password=password,
+        cert_name=certificate,
+        continuous=continuous
+    )
+    _LOGGER.info('Initializing communication with host %s (username: %s)',
+                 host, username)
+    yield from hass.async_add_job(roomba.connect)
+    roomba_vac = RoombaVacuum(name, roomba)
+    hass.data[PLATFORM][host] = roomba_vac
+
+    async_add_devices([roomba_vac], update_before_add=True)
+
+
+class RoombaVacuum(VacuumDevice):
+    """Representation of a Xiaomi Vacuum cleaner robot."""
+
+    def __init__(self, name, roomba):
+        """Initialize the Roomba handler."""
+        self._available = False
+        self._battery_level = None
+        self._is_on = False
+        self._name = name
+        self._state_attrs = {}
+        self._status = None
+        self.vacuum = roomba
+        self.vacuum_state = None
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_ROOMBA
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the vacuum cleaner."""
+        return self._battery_level
+
+    @property
+    def status(self):
+        """Return the status of the vacuum cleaner."""
+        return self._status
+
+    @property
+    def state(self) -> str:
+        """Return the state."""
```

Comment:
No need to replicate here the `state` property as you're implementing the `is_on` prop